exonum-java-proofs/src/main/java/com/exonum/binding/storage/proofs/map/flat/UncheckedFlatMapProof.java

```diff
@@ -0,0 +1,176 @@
+package com.exonum.binding.storage.proofs.map.flat;
+
+import static com.exonum.binding.hash.Funnels.hashCodeFunnel;
+import static com.exonum.binding.storage.proofs.DbKeyFunnel.dbKeyFunnel;
+
+import com.exonum.binding.hash.HashCode;
+import com.exonum.binding.hash.HashFunction;
+import com.exonum.binding.hash.Hashing;
+import com.exonum.binding.storage.proofs.map.DbKey;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * An unchecked flat map proof, which does not include any intermediate nodes.
+ */
+public class UncheckedFlatMapProof implements UncheckedMapProof {
+
+  private static final HashFunction HASH_FUNCTION = Hashing.defaultHashFunction();
+
+  private final List<MapProofEntry> proofList;
+
+  private final Set<byte[]> requestedKeys;
+
+  UncheckedFlatMapProof(List<MapProofEntry> proofList, Set<byte[]> requestedKeys) {
```

Comment:
The unchecked map proof already contains some keys for which it provides a proof as MapProofEntries. It shall not carry the _requested_ keys — it is the caller of the map validator who supplies them.

Reply:
Unchecked map proof won't contain absent keys. Without requested keys it contains only list of MapProofEntries and from it we have no reliable way to know what keys were requested. Plus we can't derive missing keys without knowing what keys were requested.